### PR TITLE
Use carto-auth in pydeck-carto

### PR DESCRIPTION
**Pull request by Jesus89**
#### Change List
- Use carto-auth as a dependency (remove the CartoAuth code)
- Review tests, examples and docs.

Note: merge this PR https://github.com/visgl/deck.gl/pull/7255 first.

### CODE REVIEW

**Comment by Jesus89** [comment]
Released:

carto-pydeck: 0.1.0b1

**Comment by ibgreen** [comment]
The CARTO module is obviously unique right now, it is a company specific module that is actively being developed as open source directly in the vis.gl repos, and that is of course very cool. 

So it is not surprising that it is pushing some of our existing conventions, in this case by being the first module to abandon our long standing module naming conventions.

By that I mean that for all vis.gl javascript modules, we have been using the `@deck.gl/` organization prefix on all module names to signal that these are legit packages, guaranteed to be published by the vis.gl team. No one else can publish a module starting with `@deck.gl/...` and the name prefix shows that the package is officially supported by vis.gl community.

My assumption has been that we would follow similar conventions in Python (and to date we have been using the `pydeck-` prefix). 

Obviously, this PR breaks this pattern,. While I can understand the wish to do so from a marketing perspective, by making this change, we are losing some things:

- we dilute the implicit promise and easy recognizability of existing vis.gl module naming schemes.

- we open for other integration modules to request make the same change, e.g. put their company names first. Say `google-earthengine-pydeck` and by extension `@mapbox/deck.gl` etc. It would now be harder to say no to such requests, once we have this precedent.

Somehow the new name is signaling that this is a primarily a CARTO owned and published module rather than a vis.gl owned and published integration module. As much as I appreciate the carto module being developed in the open in the deck.gl repo, one could argue that the source code for something named carto-pydeck should perhaps not be located in the deck.gl repo. 

I am not hard opposed to this PR, but I did want to state my concerns (and TSC should be a great forum if we did want to discuss how to balance such things).

**Comment by ibgreen** [comment]
If we wanted to formalize python package naming (good idea regardless of whether we proceed with this "exception") we could look into the name scoping semantics in the Python package managers. I suspect that @kylebarron has sufficiently deep understanding of how things work here (e.g. `pydeck.modulename`)

**Comment by kylebarron** [comment]
> No one else can publish a module starting with `@deck.gl/...` and the name prefix shows that the package is officially supported by vis.gl community.

One note: the Python Package Index doesn't support 
_reserving_ a namespace in the same way NPM does. So in Python, we could adopt a `pydeck` namespace, but it wouldn't prevent third parties from _also_ publishing a `pydeck`-namespaced module.

**Comment by Jesus89** [comment]
Hi @ibgreen and @kylebarron.

I totally get your point. The motivation for the change was way simpler: align with the new `carto-auth` package naming, but I agree with the semantics of using `pydeck-*` as a general naming rule, although, as Kyle mentions, anyone can publish a Python package with that name template.

For this pull request, I will update it to revert the name change but keep the code, test, and examples.

**Comment by kylebarron** [comment]
> but I agree with the semantics of using `pydeck-*` as a general naming rule, although, as Kyle mentions, anyone can publish a Python package with that name template.

This is true, though I was also referring to how it would work if pydeck were published as a namespace package. So existing pydeck would change to something like `import pydeck.core`, and then new pydeck extensions could be distributed under the same namespace, like `import pydeck.carto`, etc

**Comment by Jesus89** [comment]
Regarding the public documentation.

I've updated both pydeck and pydeck-carto ones. Now pydeck points to this repository instead of the old Uber one. Config file `.readthedocs.yml` is not required anymore.

- https://deckgl.readthedocs.io/
- https://pydeck-carto.readthedocs.io/

File: bindings/pydeck-carto/docs/layer.rst [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+* **data**: (``str``) Either a SQL query or a name of dataset/tileset.
+
+* **type_**: (``pydeck.types.String``) Type of the input data. It can be either QUERY, TABLE or TILESET. :meth:`pydeck_carto.layer.MapType`.
+
+* **connection**: (``pydeck.types.String``) Name of the connection registered in the CARTO Workspace. The connection for the CARTO Data Warehouse is already defined as a constant :meth:`pydeck_carto.layer.CartoConnection`.
+
+* **credentials**: (``dict``) Defines the app credentials to gather the information from CARTO. It is recommended to use :meth:`pydeck_carto.get_layer_credentials` to obtain automatically the token from Oauth using the `carto-auth package <https://github.com/cartodb/carto-auth>`_.
+
+* **geo_column**: (``pydeck.types.String``, optional) Name of the geo_column in the CARTO platform. It also support spatial indexes (h3, quadbin) :meth:`pydeck_carto.layer.GeoColumnType`.
```
Comment (felixpalmer) [inline bindings/pydeck-carto/docs/layer.rst:52]: Are the aggregation functions missing for a reason?

File: bindings/pydeck-carto/docs/authentication.rst [hunk 2]
```diff
@@ -0,0 +1,34 @@
+Authentication
+==============
+
+Pydeck-carto supports two types of authentications using the `carto-auth <https://github.com/cartodb/carto-auth>`_ package.
```
Comment (felixpalmer) [inline bindings/pydeck-carto/docs/authentication.rst:4]:
```suggestion
Pydeck-carto supports two types of authentication using the `carto-auth <https://github.com/cartodb/carto-auth>`_ package.
```

**Review by felixpalmer — APPROVED** [review]
Looking nice, great that you've added more info on how to get setup

File: bindings/pydeck-carto/docs/layer.rst (hunk 1 above)
Comment (Jesus89) [reply bindings/pydeck-carto/docs/layer.rst:52]: Nope, actually, I think it should link to the CartoLayer documentation in deck.gl

**Pull request merged**